app/shop.py

```diff
@@ -0,0 +1,39 @@
+from __future__ import annotations
+
+import datetime
+from dataclasses import dataclass
+from math import ceil
+
+
+@dataclass

```

Comment:
It looks to me that this class has a little to much in the way of methods to be a dataclass. I might be wrong.